packages/extension-vscode/src/extension.ts

```diff
@@ -0,0 +1,53 @@
+import { workspace, ExtensionContext } from 'vscode';
+
+import {
+    LanguageClient,
+    LanguageClientOptions,
+    ServerOptions,
+    TransportKind
+} from 'vscode-languageclient';
+
+// List of document types the extension will run against.
+const supportedDocuments = [
+    'css',
+    'html',
+    'javascript',
+    'json',
+    'jsonc'
+];
+
+// Keep a reference to the client to stop it when deactivating.
+let client: LanguageClient;
+
+export const activate = (context: ExtensionContext) => {
+
+    const serverModule = context.asAbsolutePath('dist/src/server.js');
+
+    const serverOptions: ServerOptions = {
+        debug: {
+            module: serverModule,
+            options: { execArgv: ['--nolazy', '--inspect=6009'] },
+            transport: TransportKind.ipc
+        },
+        run: {
+            module: serverModule,
+            transport: TransportKind.ipc
+        }
+    };
+
+    const clientOptions: LanguageClientOptions = {
+        documentSelector: supportedDocuments,
+        synchronize: {
+            // Notify the server if a webhint-related configuration changes
+            fileEvents: workspace.createFileSystemWatcher('**/.hintrc')
+        }
+    };
+
+    // Create and start the client (also starts the server)
+    client = new LanguageClient('webhint', serverOptions, clientOptions);
+    client.start();
+};
+
+export const deactivate = (): Thenable<void> => {
```

Comment:
Ah, I usually prefer `Promise`, but `Thenable` is a subset of `Promise` that just includes `.then()`. The examples used it likely as a more generic option (as it's more cross-promise-implementation-compatible, but that doesn't really matter in this context). I'll track down any remaining ones and convert them all to `Promise`.

Reply:
I took a look and `client.stop()` (which comes from `vscode-languageclient`) is returning a `Thenable`. This is why `Thenable` is being used here. It cannot be cast to a `Promise` since `Thenable` is a subset so this one should remain as-is.